jenkins/Jenkinsfile

```diff
@@ -0,0 +1,256 @@
+                sh '''
+                #!/bin/bash
+                set -eux
+                git clone https://github.com/yyuu/pyenv.git ~/.pyenv
+                export PYENV_ROOT="$HOME/.pyenv"
+                export PATH="$PYENV_ROOT/bin:$PATH"
+                echo 'export PATH="$HOME/.pyenv/bin:$PATH"' >> ~/.bashrc
+                sudo apt-get install -y build-essential libssl1.0-dev zlib1g-dev xz-utils
+                echo 'if [ -e $SHLVL = 1 ]; then
+                  echo 'eval "$(pyenv init -)"' >> ~/.bashrc
+                  echo 'eval "$(pyenv virtualenv-init -)"' >> ~/.bashrc
+                  fi'
+                pyenv install 2.6.9
+                pyenv global 2.6.9
+                '''     
+                install_test_dependencies('py26')
+                pytest('dsl_parser')
+                pytest('script_runner')
+                pytest('cloudify')
+                /* store test results in path: test-results
+                */
+              }
+            }
+          }
+        }
+        stage('test_py27'){
+          steps{
+            sh "mkdir -p ${env.WORKSPACE}/test-py27 && cp -rf ${env.WORKSPACE}/${env.PROJECT}/. ${env.WORKSPACE}/test-py27"
+            container('py27'){
+              dir("${env.WORKSPACE}/test-py27") {
+                install_test_dependencies('py27')
+                pytest('dsl_parser')
+                pytest('script_runner')
+                pytest('cloudify')
+                /* storte test results in path: test-results */
+              }
+            }
+          }
+        }
+        stage('test_py36'){
+          steps{
+            sh "mkdir -p ${env.WORKSPACE}/test-py36 && cp -rf ${env.WORKSPACE}/${env.PROJECT}/. ${env.WORKSPACE}/test-py36"
+            container('py36'){
+              dir("${env.WORKSPACE}/test-py36") {
+                install_test_dependencies('py36')
+                pytest('dsl_parser')
+                pytest('script_runner')
+                pytest('cloudify')
+                /* store test results in path: test-results */
+              }
+            }
+          }
+        }
+      }
+    }
+    stage('build_all_project_docs'){
+      when { branch pattern: ".+-build", comparator: "REGEXP"}
+      steps{
+        sh "mkdir -p ${env.WORKSPACE}/project-docs && cp -rf ${env.WORKSPACE}/${env.PROJECT}/. ${env.WORKSPACE}/project-docs"
+        container('py27'){
+          dir("${env.WORKSPACE}/project-docs") {
+            /* restore cache with relevant keys */
+            echo 'Install sphinx'
+            sh "pip install sphinx sphinx-rtd-theme"
+            echo 'Install cloudify-common'
+            sh "pip install ."
+            build_docs('cloudify_plugins_common')
+            build_docs('cloudify_rest_client')
+            build_docs('cloudify_script_plugin')
+            build_docs('cloudify_dsl_parser')
+            /* save cache */
+          }
+        }
+      }
+    }
+    stage('deploy_all_project_docs'){
```

Comment:
the step looks really bad inside the jenkins, can we wrap it with add `echo` with a pretty message like what folder is uploaded to what s3 path?

https://jenkins.eks.cloudify.co/blue/organizations/jenkins/Common-Multibranch/detail/yuval-build/3/pipeline